FeedStoreChallenge/CoreData/CoreDataFeedStore.swift

```diff
@@ -29,14 +29,54 @@ public final class CoreDataFeedStore: FeedStore {
 	}
 
 	public func retrieve(completion: @escaping RetrievalCompletion) {
-		fatalError("Must be implemented")
+		perform { context in
+			do {
+				if let cache = try ManagedCache.find(in: context) {
+					completion(.found(feed: cache.localFeed,
+					                  timestamp: cache.timestamp))
+				} else {
+					completion(.empty)
+				}
+			} catch {
+				completion(.failure(error))
+			}
+		}
 	}
 
 	public func insert(_ feed: [LocalFeedImage], timestamp: Date, completion: @escaping InsertionCompletion) {
-		fatalError("Must be implemented")
+		perform { context in
+			do {
+				let feed = ManagedFeedImage.images(from: feed, in: context)
+				let cache = try ManagedCache.newUniqueInstance(in: context)
+				cache.timestamp = timestamp
+				cache.feed = feed
```

Comment:
Suggestion:
There's no need to convert `feed` to images before `try ManagedCache.newUniqueInstance(in: context)` because try can fail and unnecessary work can be performed.
Also setting `cache.feed` can be a one-liner.

```suggestion
				let cache = try ManagedCache.newUniqueInstance(in: context)
				cache.timestamp = timestamp
				cache.feed = ManagedFeedImage.images(from: feed, in: context)
```